### DISCUSSION TITLE
ACE 3 - BODYBAG Functions

### DISCUSSION THREAD

**Discussion by thecosmicfighter** [acemod/ACE3#10079] [Ideas]
I am currently creating a multiplayer server where players can choose their jobs.

I wanted to submit the idea of ​​being able to remove dead players or NPCs from their body bags, this would improve the realism for role-playing games.

Is this technically possible ?

**Comment by PabstMirror** [comment]
Once a body goes into a bodybag it's gone
(there was a old PR https://github.com/acemod/ACE3/pull/3316 that would make bodybags vehicles, but it had some downsides)

You could probably do something like
- save some info about body when placing into a bodybag
- add an action on the body bag to create a dead guy and restore that info

Really rough starting point would be something like this:
```
["ace_placedInBodyBag", {
    params ["_target", "_bodyBag", "_isGrave"];
    if (_isGrave) exitWith {};
    // save uniform to bodybag object?
}] call CBA_fnc_addEventHandler;

private _removeDead = [
    "xx_removeDead",
    "take body",
    "",
    {
        // delete bodybag
        // create dead guy at same place
        // restore uniform?
    },
    {true} // ?
] call EFUNC(interact_menu,createAction);
["ACE_bodyBagObject", 0, ["ACE_MainActions"], _removeDead, true] call EFUNC(interact_menu,addActionToClass);
```

**Reply by thecosmicfighter** [reply]
Thanks. 

It seems a little complicated for me as I am a beginner. 

I'll try to do something with it

**Reply by thecosmicfighter** [reply]
@PabstMirror when you say "save some info about body" what kind of info could it be ?
